examples/fluids/qfunctions/stg_shur14.h

```diff
@@ -274,70 +303,79 @@ CEED_QFUNCTION(STGShur14_Inflow)(void *ctx, CeedInt Q,
   const CeedScalar cp     = stg_ctx->newtonian_ctx.cp;
+  const CeedScalar Rd     = cp - cv;
   const CeedScalar gamma  = cp/cv;
 
-  CeedPragmaSIMD
-  for(CeedInt i=0; i<Q; i++) {
-    const CeedScalar rho        = q[0][i];
-    const CeedScalar u[]        = {q[1][i]/rho, q[2][i]/rho, q[3][i]/rho};
-    const CeedScalar E_kinetic  = .5 * rho * (u[0]*u[0] + u[1]*u[1] + u[2]*u[2]);
-    const CeedScalar E_internal = q[4][i] - E_kinetic;
-    const CeedScalar P          = E_internal * (gamma - 1.);
+  const CeedScalar theta0 = stg_ctx->theta0;
+  const bool prescribe_T  = stg_ctx->prescribe_T;
 
-    const CeedScalar wdetJb  = (is_implicit ? -1. : 1.) * q_data_sur[0][i];
-    // ---- Normal vect
+  CeedPragmaSIMD
+  // Quadrature Point Loop
+  for (CeedInt i=0; i<Q; i++) {
+    // Setup
+    // Setup
+    // -- Interp-to-Interp q_data
+    // For explicit mode, the surface integral is on the RHS of ODE q_dot = f(q).
+    // For implicit mode, it gets pulled to the LHS of implicit ODE/DAE g(q_dot, q).
+    // We can effect this by swapping the sign on this weight
+    const CeedScalar wdetJb  = (implicit ? -1. : 1.) * q_data_sur[0][i];
+
+    // Calculate inflow values
+    CeedScalar velocity[3];
+    for (CeedInt j=0; j<3; j++) velocity[j] = jac_data_sur[5+j][i];
+
+    // enabling user to choose between weak T and weak rho inflow
+    CeedScalar drho, dE, dP;
+    if (prescribe_T) {
+      // rho should be from the current solution
+      drho = dq[0][i];
+      CeedScalar dE_internal = drho * cv * theta0;
+      CeedScalar dE_kinetic = .5 * drho * Dot3(velocity, velocity);
+      dE = dE_internal + dE_kinetic;
+      dP = drho * Rd * theta0; // interior rho with exterior T
+    } else { // rho specified, E_internal from solution
+      drho = 0;
+      dE = dq[4][i];
+      dP = dE * (gamma - 1.);
+    }
     const CeedScalar norm[3] = {q_data_sur[1][i],
                                 q_data_sur[2][i],
                                 q_data_sur[3][i]
                                };
 
-    const CeedScalar E = E_internal + E_kinetic;
+    const CeedScalar u_normal = Dot3(norm, velocity);
 
-    // Velocity normal to the boundary
-    const CeedScalar u_normal = norm[0]*u[0] +
-                                norm[1]*u[1] +
-                                norm[2]*u[2];
-    // The Physics
-    // Zero v so all future terms can safely sum into it
-    for (CeedInt j=0; j<5; j++) v[j][i] = 0.;
-
-    // The Physics
-    // -- Density
-    v[0][i] -= wdetJb * rho * u_normal;
-
-    // -- Momentum
-    for (CeedInt j=0; j<3; j++)
-      v[j+1][i] -= wdetJb *(rho * u_normal * u[j] +
-                            norm[j] * P);
-
-    // -- Total Energy Density
-    v[4][i] -= wdetJb * u_normal * (E + P);
-  }
+    v[0][i] = - wdetJb * drho * u_normal;
+    for (int j=0; j<3; j++)
+      v[j+1][i] = -wdetJb * (drho * u_normal * velocity[j] + norm[j] * dP);
+    v[4][i] = - wdetJb * u_normal * (dE + dP);
+  } // End Quadrature Point Loop
```

Comment:
@LeilaGhaffari Heads up about this incoming code. If we use `FluxInviscid_fwd`, it'll help make it match what you're doing with primitive variables.